recipes/open62541/all/conanfile.py

```diff
@@ -216,9 +216,9 @@ def validate(self):
                     "Lower Open62541 versions than 1.1.0 are not cpp compatible due to -fpermisive flags")
 
         max_clang_version = "8" if Version(self.version) < "1.1.0" else "9"
-        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) > max_clang_version:
+        if self.settings.compiler == "clang" and Version(self.version) < "1.3.8" and Version(self.settings.compiler.version) > max_clang_version:
             raise ConanInvalidConfiguration(
-                "Open62541 supports Clang up to {} compiler version".format(max_clang_version))
+                "Open62541 version {} does not support Clang version {}".format(Version(self.version),Version(self.settings.compiler.version)))
```

Comment:
Do you know if it's just 1.3.8 onwards that adds support for newer clang versions? From what you said in the issue, I'd guess that if it's still relevant at all, it's only for much older versions.

Reply:
My thinking was that this way the package does not change for older versions, i.e. it is "backwards compatible". 

In generell I see no reason to only allow versions >= 1.3.8. We could just remove this check and see what happens. I just do not want to debug old versions of the package.

Reply:
Adding support for more versions of clang only adds users, so it's backwards compatible as is, but it's preferable to have as many valid configurations as possible.

I can check a few clang versions locally and see what happens. But also, I think some of the patch revisions can be removed, currently there's:
```
1.3.8
1.3.6
1.3.4
1.3.1
1.2.6
1.2.4
1.2.2
1.1.6
1.1.5
1.1.3
1.0.6
1.0.3
```
I don't know how much changes between versions in open62541, but I would think that shortening the list to just the latest patch version in each minor version would be sufficient:
```
1.3.8
1.2.6
1.1.6
1.0.6
```
@RubenRBS Thoughts on this?

Reply:
Ok, if we can remove versions I am all for it. It takes for ever to build all of these packages during a PR.

Reply:
I already made this change (I really want to be done with this). If this change is not needed we can always revert to commit `21c656d91905b5ebe971790e5c1d2f4d8f97f969`.

Reply:
I'm having difficult building the package at all locally, I'll try to test those clang versions tomorrow.

Reply:
I can't seem to build in any configuration locally, unsure what the issue is. Having those clang checks for only later versions is probably fine for now.

Reply:
It should work now. I used a patch to get rid of the sanitizer.